syntax = "proto3";

package nfa_msg;

service Runtime_RPC {
  // Sends a greeting
  rpc LivenessCheck (LivenessRequest) returns (LivenessReply) {}
  rpc AddOutputRts (AddOutputRtsReq) returns (AddOutputRtsRes) {} 
}

message LivenessRequest {

}

message LivenessReply {

}

message RuntimeConfig{
  uint64 worker_id = 1;
  string input_port_mac = 2;
  string output_port_mac = 3;
  string control_port_mac = 4;
  string rpc_ip = 5;
  uint32 rpc_port = 6;
}

message AddOutputRtsReq{
  message Address{
    string rpc_ip = 1;
    uint32 rpc_port = 2;
  }
  repeated Address addrs = 1;
}

message AddOutputRtsRes{
  repeated RuntimeConfig rt_configs = 1;
}
